Standards/scs-0217-v1-cluster-hardening.md

```diff
@@ -0,0 +1,470 @@
+of the clusters' user (and therefore not possible to implement for the provider), being more relevant
+for high security clusters or changing the expected cluster environment too much, so that normally
+expected operations could potentially not work in such a modified cluster.
+These measurements will possibly be introduced in a future document about higher security clusters.
+
+## Standard
+
+This standard provides the baseline security requirements for a cluster in the SCS context. It tries
+to increase the security in contrast to a non-configured setup by requiring and recommending specific
+settings that provide different security improvements without being too intrusive.
+
+Kubernetes clusters MUST be updated regularly in order to receive bugfixes and security patches.
+For more information refer to the [SCS K8s Version Policy](scs-0210-v2-k8s-version-policy.md),
+which outlines the update policies of the SCS.
+
+Hardening etcd is important due to it being a critical component inside a Kubernetes cluster.
+etcd SHOULD be isolated from the Kubernetes cluster by being hosted on separate (virtual) machines.
+If this is the case, access to these instances MUST be configured, so that only the API server and
+necessary cluster components requiring access can access etcd.
+Communication with etcd MUST be secured with TLS for both peer- and cluster-communication.
+It is RECOMMENDED to use a CA separate from the one used for the Kubernetes cluster for etcd in
+order to better control and issue certificates for clients allowed to access etcd.
+ACL MUST be enabled for etcd, which allows better control of the access rights to specific key sets
+for specific users. Authentication MUST be done via the Common Name (CN) field of the TLS client
+certificates (since normal username-password-authentication isn't implemented for Kubernetes).
+
+Kubernetes' endpoints MUST be secured in order to provide a small attack surface for bad actors.
+It MUST NOT be possible to access Kubernetes ports from outside the internal network hosting the
+Kubernetes cluster except for the ports of the API server (default 6443) and the NodePort Services
+(default 30000-32767). It is also possible to give access to the read-only Kubelet API port (default 10255),
+which is mostly used for monitoring, but we advise against this, mainly because the port is HTTP-only
+and can deliver sensitive information to the outside. It is RECOMMENDED to disable this port altogether,
+if it isn't in use. Endpoints MUST be secured via HTTPS.
+
+Securing Kubernetes via authentication and authorization is another important topic here.
+Authentication is possible through multiple mechanisms, including Kubernetes-provided systems as well as external
+authentication processes.
+A cluster MUST implement at least two methods for authentication. One of these MUST be *Service Account Tokens*, in order
+to provide full functionality to Pods. A second authentication mechanisms can be chosen depending
```

Comment:
I think the end of this sentence is missing? depending on what?

Reply:
Updated the sentence in [6805ea0](https://github.com/SovereignCloudStack/standards/pull/581/commits/6805ea0289d5595323c73ed5df8b29d9e90fd889)